### PR TITLE
Profile settings page

### PR DESCRIPTION
**Pull request by onkar-saeloun**
## Notion card
https://www.notion.so/saeloun/Redesign-Profile-settings-page-Integration-03e66940e98d4e39935108562dfedcff

## Summary

- Added page design for profile settings.
- Added YUP validation for the necessary fields.
- API integration is added.
- Added Left sidebar for the Profile settings and rest of the settings pages.

## Preview

https://user-images.githubusercontent.com/104751221/172360268-0e3ed269-4e16-419b-b99c-89e05c80d8bf.mp4

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested Manually at my machine.
Testing involves following things-

- Update button will be enabled only by making changes in any field otherwise it should be disabled.
- Cancel button should remove all the unsaved changes.
- Firstname, Lastname should be mandatory fields. After clicking on change password button, (current password, password and confirm password) will become mandatory fields.
- password and confirm password should match on UI before submitting.
- after deleting the profile image should be deleted.
- changes should be updated after refreshing the page.

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by supriya3105** [comment]
@harshaanNihal @apoorv1316 Please review and test this PR respectively.

File: app/controllers/internal_api/v1/profile_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  def index
+    authorize :index, policy_class: ProfilePolicy
+    if current_user.avatar.attached?
+      avatar_url = url_for(current_user.avatar)
+    end
+    render json: { user: current_user.as_json.merge("avatar_url" => avatar_url) }, status: :ok
+  end
+
+  def remove_avatar
+    authorize :remove_avatar, policy_class: ProfilePolicy
```
Comment (harshaanNihal) [inline app/controllers/internal_api/v1/profile_controller.rb:13]:
```suggestion
    authorize :profile
```

NIT: require same changes for update & index method

File: app/controllers/internal_api/v1/profile_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class InternalApi::V1::ProfileController < InternalApi::V1::ApplicationController
```
Comment (harshaanNihal) [inline app/controllers/internal_api/v1/profile_controller.rb:3]: Please Add Specs for Profile Controller

File: app/javascript/src/components/Profile/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+const Header = ({ title, subTitle, showButtons = false, cancelAction = function (){0;}, saveAction = function (){0;}, isDisableUpdateBtn = false, updateMsg={ message: "", type: "" } }) => (
```
Comment (JijoBose) [inline app/javascript/src/components/Profile/Header.tsx:3]: same

Comment (onkar-saeloun) [reply app/javascript/src/components/Profile/Header.tsx]: Thanks Jijo. Done with the change.

File: app/controllers/internal_api/v1/profile_controller.rb (hunk 1 above)
Comment (onkar-saeloun) [reply app/controllers/internal_api/v1/profile_controller.rb:13]: @abinashpa can you please help me in resolving these comments?

File: app/controllers/internal_api/v1/profile_controller.rb (hunk 2 above)
Comment (onkar-saeloun) [reply app/controllers/internal_api/v1/profile_controller.rb:3]: @abinashpa here too.

File: app/controllers/internal_api/v1/profile_controller.rb (hunk 1 above)
Comment (apoorv1316) [reply app/controllers/internal_api/v1/profile_controller.rb:13]: @onkar-saeloun These changes are to be done from my PR. You don't have to worry about it.

File: app/controllers/internal_api/v1/profile_controller.rb (hunk 2 above)
Comment (apoorv1316) [reply app/controllers/internal_api/v1/profile_controller.rb:3]: @harshaanNihal There is a separate PR for the profile API which I have raised and I have added specs there. Onkar just pulled my branch to integrate API.

**Comment by supriya3105** [comment]
@apoorv1316 Please test this PR and share your feedback if any.

**Comment by apoorv1316** [comment]
Already done on a call with Onkar. He has made all the required changes. Approving it.

**Review by apoorv1316 — APPROVED** [review]

**Comment by supriya3105** [comment]
@JijoBose Please review this PR .

**Review by JijoBose — APPROVED** [review]

**Pull request merged**